pkgs/development/compilers/dmd/generic.nix

```diff
@@ -0,0 +1,255 @@
+{ version
+, dmdSha256
+, druntimeSha256
+, phobosSha256
+}:
+
+{ stdenv
+, lib
+, fetchFromGitHub
+, makeWrapper
+, which
+, writeTextFile
+, curl
+, tzdata
+, gdb
+, Foundation
+, callPackage
+, targetPackages
+, fetchpatch
+, bash
+, installShellFiles
+, git
+, unzip
+, HOST_DMD ? "${callPackage ./bootstrap.nix { }}/bin/dmd"
+}:
+
+let
+  dmdConfFile = writeTextFile {
+    name = "dmd.conf";
+    text = (lib.generators.toINI { } {
+      Environment = {
+        DFLAGS = ''-I@out@/include/dmd -L-L@out@/lib -fPIC ${lib.optionalString (!targetPackages.stdenv.cc.isClang) "-L--export-dynamic"}'';
+      };
+    });
+  };
+
+  bits = builtins.toString stdenv.hostPlatform.parsed.cpu.bits;
+  osname =
+    if stdenv.isDarwin then
+      "osx"
+    else
+      stdenv.hostPlatform.parsed.kernel.name;
+
+  pathToDmd = "\${NIX_BUILD_TOP}/dmd/generated/${osname}/release/${bits}/dmd";
+in
+
+stdenv.mkDerivation rec {
+  pname = "dmd";
+  inherit version;
+
+  enableParallelBuilding = true;
+
+  srcs = [
+    (fetchFromGitHub {
+      owner = "dlang";
+      repo = "dmd";
+      rev = "v${version}";
+      sha256 = dmdSha256;
+      name = "dmd";
+    })
+    (fetchFromGitHub {
+      owner = "dlang";
+      repo = "druntime";
+      rev = "v${version}";
+      sha256 = druntimeSha256;
+      name = "druntime";
+    })
+    (fetchFromGitHub {
+      owner = "dlang";
+      repo = "phobos";
+      rev = "v${version}";
+      sha256 = phobosSha256;
+      name = "phobos";
+    })
+  ];
+
+  sourceRoot = ".";
+
+  # https://issues.dlang.org/show_bug.cgi?id=19553
+  hardeningDisable = [ "fortify" ];
+
+  # Not using patches option to make it easy to patch, for example, dmd and
+  # Phobos at same time if that's required
+  patchPhase = ''
+    runHook prePatch
```

Comment:
This is unexpected and we shouldn't do that. Can we move this to postPatch?

Reply:
I've moved the patches from manual patch calls with `--directory` arguments into `patches` with the `fetchpatch`-equivalent options (`stripLen` & `extraPrefix`), tested by building 2.087.0. I hope the comment was just about how to fetch & apply these with the way the `srcs` are composited (or rather not-composited), because I can't think of anything else that would require such a weird handling.

Reply:
Yes the comment was about that. I was not aware of the `extraPrefix`/`stripLen` options. Thanks - this is indeed better.